core/src/main/java/com/github/mizool/core/rest/errorhandling/GenericErrorMapper.java

```diff
@@ -31,39 +29,42 @@
 
 @Slf4j
 @VisibleForTesting
-public class ErrorMapper
+public class GenericErrorMapper
 {
     @VisibleForTesting
     public static final String GLOBAL_PROPERTY_KEY = "GLOBAL";
 
-    public ErrorResponse handleErrorAccordingToBehaviour(Throwable t, ErrorHandlingBehavior behaviour)
-    {
-        logError(t, behaviour);
-
-        Map<String, String> parameters = null;
-        if (behaviour.includeDetails())
-        {
-            parameters = createExceptionParameters(t);
-        }
+    private final ErrorHandlingBehaviorCatalog errorHandlingBehaviorCatalog;
 
-        ErrorDto error = ErrorDto.createGenericError(parameters);
-        if (behaviour.includeErrorId())
-        {
-            error = new ErrorDto(t.getClass().getName(), parameters);
-        }
-        ErrorMessageDto errorMessage = createErrorMessageDto(error);
-        return new ErrorResponse(behaviour.getStatusCode(), errorMessage);
+    public GenericErrorMapper()
+    {
+        errorHandlingBehaviorCatalog = new ErrorHandlingBehaviorCatalog();
     }
 
-    public ErrorResponse handleClientError(ClientErrorException e)
+    public ErrorResponse handleErrorAccordingToBehaviour(Throwable t)
     {
-        log.debug("Client error", e);
-        int statusCode = e.getResponse().getStatus();
-        Class<? extends Exception> errorClass = determineErrorClass(statusCode, e.getClass());
-
-        ErrorDto error = new ErrorDto(errorClass.getName(), null);
-        ErrorMessageDto errorMessage = createErrorMessageDto(error);
-        return new ErrorResponse(statusCode, errorMessage);
+        ErrorResponse result = null;
+        Optional<ErrorHandlingBehavior> behaviorOptional = errorHandlingBehaviorCatalog.lookup(t);
+        if (behaviorOptional.isPresent())
+        {
+            ErrorHandlingBehavior behaviour = behaviorOptional.get();
```

Comment:
`if-isPresent-get` is discouraged and should usually be replaced with `ifPresent()` or `ifPresentOrElse()` (JDK 9). That doesn't work here due to the write to `result`, but we can still get rid of `isPresent-get` by turning this into `ErrorResponse result = behaviorOptional.map(...).orElse(null)` (either with an inline lambda or a method reference).

Also, rename the local variable `behaviour` to `behavior`.